robot/web/web.go

```diff
@@ -498,6 +503,68 @@ func (svc *webService) runWeb(ctx context.Context, options weboptions.Options) (
 	return err
 }
 
+// Namer is used to get a resource name from incoming requests for countingfor request. Requests for
+// resources are expected to be a gRPC object that includes a `GetName` method.
+type Namer interface {
+	GetName() string
+}
+
+// RequestCounter maps string keys to atomic ints that get bumped on every incoming gRPC request for
+// components.
+type RequestCounter struct {
+	counts sync.Map
+}
+
+// UnaryInterceptor returns an incoming server interceptor that will pull method information and
+// optionally resource information to bump the request counters.
+func (mc *RequestCounter) UnaryInterceptor(
```

Comment:
should we add a stream interceptor as well?

Reply:
I think it's worthwhile, but not necessarily a blocker for this.

A stream being opened is an interesting event for sure, but we might care more about the messages being sent within a stream. And I think most stream calls are for webrtc dialing, which isn't covered by the `viam.components` prefix right now.